From d9441e168c7de2c15c488a48bae05ff5ec9da5e3 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 11:37:03 -0400
Subject: [ethernet] i40e: Remove casts of pointer to same type

Message-id: <1409658042-22689-63-git-send-email-sassmann@redhat.com>
Patchwork-id: 90258
O-Subject: [RHEL7.1 PATCH 062/281] i40e: Remove casts of pointer to same type
Bugzilla: 1091126
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Joe Perches <joe@perches.com>

Casting a pointer to a pointer of the same type is pointless,
so remove these unnecessary casts.

Done via coccinelle script:

$ cat typecast_2.cocci
@@
type T;
T *foo;
@@

- (T *)foo
+ foo

Signed-off-by: Joe Perches <joe@perches.com>
Tested-by: Kavindya Deegala <kavindya.s.deegala@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 3dbb7fd24794e39e503bc7844dae03c17d821326)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40e/i40e_ethtool.c b/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
index d2cf830..db57cce 100644
--- a/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
+++ b/drivers/net/ethernet/intel/i40e/i40e_ethtool.c
@@ -396,7 +396,7 @@ static int i40e_get_eeprom(struct net_device *netdev,
   ret_val = i40e_aq_read_nvm(hw, 0x0,
     eeprom->offset + (I40E_NVM_SECTOR_SIZE * i),
     len,
-    (u8 *)eeprom_buff + (I40E_NVM_SECTOR_SIZE * i),
+    eeprom_buff + (I40E_NVM_SECTOR_SIZE * i),
     last, NULL);
   if (ret_val) {
    dev_info(&pf->pdev->dev,
@@ -408,7 +408,7 @@ static int i40e_get_eeprom(struct net_device *netdev,
 
 release_nvm:
  i40e_release_nvm(hw);
- memcpy(bytes, (u8 *)eeprom_buff, eeprom->len);
+ memcpy(bytes, eeprom_buff, eeprom->len);
 free_buff:
  kfree(eeprom_buff);
  return ret_val;
-- 
1.7.1